x-pack/plugin/security/cli/build.gradle

```diff
@@ -16,7 +16,13 @@ dependencies {
   }
   testRuntimeOnly "com.google.guava:guava:${versions.jimfs_guava}"
   testImplementation project(":test:framework")
-testImplementation(testArtifact(project(xpackModule('core'))))}
+  testImplementation(testArtifact(project(xpackModule('core')))) {
+    // BC and BC-FIPS together causes "Jar Hell" failures in JUnit
+    exclude group: 'org.bouncycastle', module: 'bc-fips'
+    exclude group: 'org.bouncycastle', module: 'bcpg-fips'
```

Comment:
we don't have bcpg-fips in `core` right ? So this exclusion is not needed.

Reply:
bcpg-fips is part of plugin-cli with bc-fips.
```
[root@d0747cd87f3f plugin-cli]# ls -l
total 4000
-r--r--r-- 1 elasticsearch root 3775682 Oct  8  2020 bc-fips-1.0.2.jar
-r--r--r-- 1 elasticsearch root  271757 Oct  8  2020 bcpg-fips-1.0.4.jar
-r--r--r-- 1 elasticsearch root   43784 Aug 26 09:02 elasticsearch-plugin-cli-7.14.1.jar
```

elasticsearch-security-cli does not inherit bcpg-fips from core, so it is more precautionary to exclude both bcpg-fips and bc-fips in elasticsearch-security-cli.

Reply:
> bcpg-fips is part of plugin-cli with bc-fips.

Yeap, this is defined in `distribution/tools/plugin-cli/build.gradle`

```
  api "org.bouncycastle:bcpg-fips:1.0.4"
  api "org.bouncycastle:bc-fips:1.0.2"
```

I guess my point is that since `testArtifact(project(xpackModule('core')))` doesn't include `bcpg-fips`, there is need to exclude it here. As you pointed out, `bcpg-fips` is a dependency only for `plugin-clie`. We are pretty declarative and explicit in our dependency tracking, I think there is no need to proactively exclude a dependency in an unrelated package.

Reply:
I agree with @jkakavas that the exclude should not be required. How does this actually solve the problem with the security cli junit runner?

Reply:
Thank you for the clarification that we should be explicit about exclusions. I will remove the pcpg-fips exclusion.